src/main/java/it/gov/pagopa/pu/debtpositions/controller/ReceiptControllerImpl.java

```diff
@@ -38,4 +48,29 @@ public ResponseEntity<ReceiptDTO> getReceipt(Long receiptId) {
   public ResponseEntity<ReceiptDetailDTO> getReceiptDetail(Long receiptId, String operatorExternalUserId, Long organizationId) {
     return ResponseEntity.ok(receiptService.getReceiptDetail(receiptId, operatorExternalUserId, organizationId));
   }
+
+  @Override
+  public ResponseEntity<Resource> getReceiptPdf(Long receiptId, Long organizationId) {
+    ReceiptDetailDTO receiptDetail = receiptService.getReceiptDetail(receiptId, SecurityUtils.getCurrentUserExternalId(), organizationId);
+    if(receiptDetail==null){
+      throw new NotFoundException("Receipt with ID "+receiptId+" not found");
+    }
+    Organization organization = organizationService.getOrganizationById(organizationId, SecurityUtils.getAccessToken())
+      .orElseThrow(() -> new NotFoundException("Organization with ID "+organizationId+" not found"));
```

Comment:
I would move these lines on the service... the presentation layer should just invoke business logic and map to returned DTO (if different)